script/cache_critical_dns

```diff
@@ -342,21 +343,27 @@ HEALTH_CHECKS = Hash.new(
   "DISCOURSE_REDIS_HOST": lambda { |addr|
     redis_healthcheck(
       host: addr,
+      port: env_as_int("DISCOURSE_REDIS_PORT", DEFAULT_REDIS_PORT),
+      username: ENV["DISCOURSE_REDIS_USERNAME"],
       password: ENV["DISCOURSE_REDIS_PASSWORD"])},
   "DISCOURSE_REDIS_REPLICA_HOST": lambda { |addr|
     redis_healthcheck(
       host: addr,
+      port: env_as_int("DISCOURSE_REDIS_PORT", DEFAULT_REDIS_PORT),
+      username: ENV["DISCOURSE_REDIS_USERNAME"],
       password: ENV["DISCOURSE_REDIS_PASSWORD"])},
   "DISCOURSE_MESSAGE_BUS_REDIS_HOST": lambda { |addr|
     redis_healthcheck(
       host: addr,
       port: env_as_int("DISCOURSE_MESSAGE_BUS_REDIS_PORT", DEFAULT_REDIS_PORT),
+      username: ENV["DISCOURSE_MESSAGE_BUS_REDIS_USERNAME"],
       password: ENV["DISCOURSE_MESSAGE_BUS_REDIS_PASSWORD"])},
   "DISCOURSE_MESSAGE_BUS_REDIS_REPLICA_HOST": lambda { |addr|
     redis_healthcheck(
       host: addr,
       port: env_as_int("DISCOURSE_MESSAGE_BUS_REDIS_REPLICA_PORT", DEFAULT_REDIS_PORT),
-      password: ENV["DISCOURSE_MESSAGE_BUS_REDIS_PASSWORD"])},
+      username: ENV["DISCOURSE_MESSAGE_BUS_REDIS_REPLICA_USERNAME"],
+      password: ENV["DISCOURSE_MESSAGE_BUS_REDIS_REPLICA_PASSWORD"])},
```

Comment:
```suggestion
      password: ENV["DISCOURSE_MESSAGE_BUS_REDIS_PASSWORD"])},
```

Sorry one more. I just noticed that is no `message_bus_redis_replica_password` config.